packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -73,13 +73,18 @@ function createHttpClient(): HttpClient {
 
 export function getCacheVersion(
   paths: string[],
-  compressionMethod?: CompressionMethod
+  compressionMethod?: CompressionMethod,
+  crossOsEnabled = false
 ): string {
-  const components = paths.concat(
-    !compressionMethod || compressionMethod === CompressionMethod.Gzip
-      ? []
-      : [compressionMethod]
-  )
+  const components = paths
+    .concat(
+      !compressionMethod || compressionMethod === CompressionMethod.Gzip
+        ? []
+        : [compressionMethod]
+    )
+    .concat(
+      process.platform !== 'win32' || crossOsEnabled ? [] : ['windows-only']
```

Comment:
This change also means that in case of windows and cross os disabled, the existing caches would get invalidated due to version mismatch. Is that correct?

Reply:
Yes. There would be cache invalidation

Reply:
And if that is acceptable then probably we can remove gzip fallback to restore old caches. Can update that to account for version difference but seems like there would be too many if-else

Reply:
I was thinking the opposite in another comment. Keep the fallback but remove the version differentiation for gzip.

Reply:
We can't remove version differentiation because:
> We need different versions of cache for different compression algorithms as same compression algo should be used to compress(save) and decompress(restore)

Reply:
I agree we cannot remove version differentiation here and also that we can remove the fallback since the caches are getting invalidated anyways. However, I would just suggest we update the logic/comment to go together for consistency.

Reply:
Yes agree, we can remove the fallback.

Reply:
Agreed we can remove gzip fallback for old caches, if that's what you mean

Reply:
Done